common/src/service/IotHubConnectionString.cs

```diff
@@ -11,9 +11,8 @@
 
 namespace Microsoft.Azure.Devices
 {
-    internal sealed class IotHubConnectionString : IAuthorizationHeaderProvider, ICbsTokenProvider
+    internal sealed class IotHubConnectionString : IotHubConnectionBase
```

Comment:
The new design is that all the service  clients will only need what is there in the base class. And we will have our different auth types derive from it. The naming here is not great but I don't want to make this a daunting PR so want to keep them for another PR. Ideally, this class (because internal) can be named IotHubConnectionStringCredential". Feel free to leave your inputs.